lib/rubocop/cop/style/redundant_self.rb

```diff
@@ -86,7 +86,7 @@ def on_blockarg(node)
 
         def on_lvasgn(node)
           lhs, _rhs = *node
-          @local_variables << lhs
+          (@local_variables_scopes[node] ||= []) << lhs
```

Comment:
I personally would prefer the hash to be initialized using `Hash#new`, setting an empty array as the default value.